src/backend/distributed/executor/multi_utility.c

```diff
@@ -391,6 +399,25 @@ multi_ProcessUtility(Node *parsetree,
 	{
 		ListCell *ddlJobCell = NULL;
 
+		if (IsA(parsetree, AlterTableStmt))
+		{
+			AlterTableStmt *alterTableStatement = (AlterTableStmt *) parsetree;
+			List *commandList = alterTableStatement->cmds;
+			ListCell *commandCell = NULL;
+
+			foreach(commandCell, commandList)
+			{
+				AlterTableCmd *command = (AlterTableCmd *) lfirst(commandCell);
+				AlterTableType alterTableType = command->subtype;
+
+				if (alterTableType == AT_AddConstraint)
+				{
+					Assert(commandList->length <= 1);
```

Comment:
Since we checked that length of commandList is smaller than 1 on ErrorIfUnsupportedAlterTableStmt, we are sure list_length(commandList) = 1. Both ErrorIfUnsupportedAlterTableStmt and PlanAlterTableStmt check the length of command list in that way, should I fix them too ?